### DISCUSSION TITLE
Web search - result returned; LLM reports no access

### DISCUSSION THREAD

**Discussion by ivostoykov** [open-webui/open-webui#7529] [Issues]
# Bug Report

## Installation Method

```
podman run -d --gpus all --network=pasta:-t,3000:8080,-u,auto,-T,auto,-U,auto --name owui \
    -v $HOME/.open-webui/backend_data:/app/backend/data \
    -v $HOME/.open-webui/logs:/var/log/open-webui \
    --restart always ghcr.io/open-webui/open-webui:main
```

## Environment

- **Open WebUI Version:** [v0.4.7 (latest)]
- **Ollama (if applicable):** [0.4.6]

- **Operating System:** [Ubuntu 24.04.1 LTS]
- **Browser (if applicable):** [Version 131.0.6778.85 (Official Build) (64-bit)]

**Web Search Settings:** `duckduckgo` Web search enabled in the settings and in the chat.

**Confirmation:**

- [ x] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ x] I have included the browser console logs.
- [ x] I have included the Podman container logs.
- [ x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

return web search result of the prompt: `search the web about podman and count the results returned`

## Actual Behavior:

LLM response: `Unfortunately, I don't have access to your query or any external sources`

## Description

From the Chrome DevTool Console, it is visible that there is a response.
**Bug Summary:**
Prompt: `search the web about podman and count the results returned`
LLM response: 
```
llama3.2:latest 11:33 pm
Searched 5 sites
Unfortunately, I don't have access to your query or any external sources, as you haven't provided a link or context for the user's query. Could you please provide more information about what you'd like to know regarding "Podman"? What aspects of Podman are you interested in? Is it related to its usage, installation, or something else?
```

Web search response (DevTool console:
```
{"sources": [{"source": {"collection_name": "3775a84d6ea3709ba3251cd9c54aba157d8849cfe709a8c5d3a818d65b8f847", "name": "podman usage guides", "type": "web_search_results", "urls": ["https://podman.io/docs", "https://devopscube.com/podman-tutorial-beginners/", "https://docs.podman.io/en/latest/Tutorials.html", "https://medium.com/@williamwarley/unleashing-podman-the-ultimate-guide-to-revolutionizing-your-container-management-3a5bdbbd5ef8", "https://phoenixnap.com/kb/podman-tutorial"]}, "document": ["Getting Started with Podman | Podman", "Getting Started with Podman | Podman", "To see the full Podman manual, use the man command:\n\nPodman Images\nImage building and management in Podman is performed with Buildah, an image-building tool that uses a lower-level coreutils interface. When a user executes a podman command related to images, Podman calls Buildah to perform the action. \nThe following sections demonstrate some important image management commands. \nSearching Images\nSearch registries for available images using the search command. \npodman search [search-term]\nLimit the number of results with the --limit option. For example, to produce three results for the \"centos\" search term, type:\npodman search --limit 3 centos\n\nInspect Image\nPrior to pulling a Podman image, it is good practice to inspect it. Use the podman inspect command and the system outputs image metadata, such as file size, architecture, OS, etc.\npodman inspect [repository or image ID]"], "metadata": [{"description": "Podman logoPodman logo", "embedding_config": "{\"engine\": \"ollama\", \"model\": \"all-minilm:l6-v2\"}", "language": "en", "source": "https://podman.io/docs", "start_index": 5, "title": "Getting Started with Podman | Podman"}, {"description": "Podman logoPodman logo", "embedding_config": "{\"engine\": \"ollama\", \"model\": \"all-minilm:l6-v2\"}", "language": "en", "source": "https://podman.io/docs", "start_index": 5, "title": "Getting Started with Podman | Podman"}, {"description": "This Podman tutorial will help you understand how Podman works and explain its command syntax, along with examples of the important commands.", "embedding_config": "{\"engine\": \"ollama\", \"model\": \"all-minilm:l6-v2\"}", "language": "en-US", "source": "https://phoenixnap.com/kb/podman-tutorial", "start_index": 7609, "title": "Podman Tutorial: How to Work with Images, Containers and Pods"}], "distances": [0.24014270305633545, 0.2764841318130493, 0.32625789729999133]}]}
```

## Reproduction Details

**Steps to Reproduce:**
1. Run the above podman command to create the instance
2. Open settings and enable web search. Set `duckduckgo` as a search engine and save
3. Open a new chat and execute any web search-related prompt

## Logs and Screenshots

**Browser Console Logs:**
[chrome_console.log](https://github.com/user-attachments/files/17970813/chrome_console.log)

**Docker Container Logs:**
[owui.log](https://github.com/user-attachments/files/17970814/owui.log)

**Screenshots/Screen Recordings (if applicable):**
![prompt](https://github.com/user-attachments/assets/2b3d2edb-a02b-4f59-a82b-46f97fdd015b)
![settings](https://github.com/user-attachments/assets/bcb98015-bba9-44cd-ae2b-0defc074e920)

## Additional Information

The same behaviour is reproduced when replace duckduckgo with searxng (in another container)
The searxng instance is created with this command:
```
podman run -d --name searxng \
	--network=pasta:-t,7999:8080,-u,auto,-T,auto,-U,auto \
	-v $HOME/.open-webui/searxng:/etc/searxng \
	-v $HOME/.open-webui/logs:/var/log/uwsgi \
	--restart unless-stopped \
	docker.io/searxng/searxng:latest
```
From the searxng log is visible that the request is successful and a json is returned. LLM response is though the same - no access to external resources.

Thank you

**Comment by tjbck** [comment]
![image](https://github.com/user-attachments/assets/ddbc2849-396e-4484-ab92-9e8fa4c00d16)


![image](https://github.com/user-attachments/assets/fddc82ff-f3ae-44f5-a735-53f468684208)

Unable to reproduce the issue, keep us updated!

**Reply by ivostoykov** [reply]
Hi @tjbck ,

Thank you for your fast reaction. Within the same configuration here are a few more prompts that does not work:

I appreciate your prompt is working but below is visible that these URLs are the only returned:

![Screenshot from 2024-12-02 01-23-34](https://github.com/user-attachments/assets/33c74ada-0759-4b94-b2f5-4cb3a04e6d6f)

**Reply by tjbck** [reply]
My educated guess here would be you have a misconfigured embedding model settings, so the system is unable to retrieve the relevant information. Continue your troubleshooting journey and keep us updated!

**Reply by ivostoykov** [reply]
thanks @tjbck from the DevTool I could spot that duck2go returns `202 Ratelimit` which seems to be intended behaviour (discussed [here](https://github.com/open-webui/open-webui/discussions/6624) as well). Switched to searxng and all good. Will try later the proposed python solution for duck2go

**Comment by ivostoykov** [comment]
Thanks a lot for the support and fast responses :-)

**Comment by spammenotinoz** [comment]
Hmm..  I also get this on models where I did not update the system prompt to tell the LLM it has access to the internet.
